### PR TITLE
Features/transaction history

### PR DESCRIPTION
**Pull request by nadiia-tsytsylina**
## Describe your changes

- Firstly I created separate transaction-history folder with controller, sevice and module, but there was a cycle dependencies between Appoinment, Payment and Transaction-history services.
- So I moved all transaction-history logic to Payment;
- Updated Payment controller;
- Created TransactionHistory Table;
- Added Get all transaction by User id endpoint;
- Added createSeekerCaregiverTransactions function for automaticaly creation of transactions for seeker and caregiver after all payment operations between them.

image

image

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://homecareaid.atlassian.net/browse/CC-191)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend

- [x] Swagger documentation updated
- [x] Database requests are optimized and not redundant
- [ ] Unit tests written
- [x] use ConfigService instead of process.env
- [x] use transactions if there is a call chain that mutates data in different tables
- [ ] use @index decorator for frequently requested data

### CODE REVIEW

**Review by Simpsonkk — APPROVED** [review]

File: src/common/entities/transaction-history.entity.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  })
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  @ApiProperty({
+    example: '1e3a4c60-94aa-45de-aad0-7b4a49017b1f',
+    description: 'User ID associated with the transaction',
+  })
+  @Column()
+  userId: string;
```
Comment (lyudad) [inline src/common/entities/transaction-history.entity.ts:29]: you have userId from appointment

Comment (nadiia-tsytsylina) [reply src/common/entities/transaction-history.entity.ts:29]: yes, but appointment Id is not required. There can be transactions, that aren't connected to appointment

Comment (lyudad) [reply src/common/entities/transaction-history.entity.ts:29]: give me please example of such transaction

Comment (nadiia-tsytsylina) [reply src/common/entities/transaction-history.entity.ts:29]: Withdrawal for Caregiver and Top-up for Client.
Снимок экрана 2024-01-04 125943
Снимок экрана 2024-01-04 125956

**Pull request merged**